src/lib/functions/synchronous.js

```diff
@@ -36,14 +38,26 @@ const handleSynchronousFunction = function (err, result, response) {
   response.end()
 }
 
-const formatLambdaLocalError = (err) => `${err.errorType}: ${err.errorMessage}\n  ${err.stackTrace.join('\n  ')}`
+const formatLambdaLocalError = (err) =>
+  JSON.stringify({
+    errorType: err.errorType,
+    errorMessage: err.errorMessage,
+    trace: err.stackTrace,
+  })
 
 const handleErr = function (err, response) {
+  const errorTemplateFile = fs.readFileSync(pathModule.join(__dirname, './templates/function-error.html'), 'utf-8')
+  const errorString = typeof err === 'string' ? err : formatLambdaLocalError(err)
+  let updatedErrorTemplate = errorTemplateFile
+
   detectAwsSdkError({ err })
 
   response.statusCode = 500
-  const errorString = typeof err === 'string' ? err : formatLambdaLocalError(err)
-  response.end(errorString)
+
+  const regexPattern = /<!--@ERROR-DETAILS-->/g
+  updatedErrorTemplate = updatedErrorTemplate.replace(regexPattern, errorString)
+
+  response.end(updatedErrorTemplate)
```

Comment:
[boulder] we should check to see if the request's `Accept` header includes `text/html` and only if it does should we return this HTML body. Otherwise we return what we had been returning previously